I see that actions with Node.js v26 are failing because of `LINK : fatal error LNK1117: syntax error in option 'opt:lldltojobs=2' [D:\a\node-addon-api\node-addon-api\test\build\binding_type_check.vcxproj]`. That was fixed in https://github.com/nodejs/node-gyp/pull/3331 and released in [node-gyp v13](https://github.com/nodejs/node-gyp/releases/tag/v13.0.0).

Now about the other failures. They seem to happen on Windows Server 2025, and this is why `D:\a\node-addon-api\node-addon-api\napi-inl.h(4770,76): error C1001: Internal compiler error. [D:\a\node-addon-api\node-addon-api\test\build\binding_custom_namespace.vcxproj]`. GitHub Copilot suggested this change to fix it

```diff
@@
 template <typename T>
 template <typename InstanceWrap<T>::InstanceGetterCallback getter,
           typename InstanceWrap<T>::InstanceSetterCallback setter>
 inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceAccessor(
     const char* utf8name, napi_property_attributes attributes, void* data) {
   napi_property_descriptor desc = napi_property_descriptor();
   desc.utf8name = utf8name;
   desc.getter = details::TemplatedInstanceCallback<T, getter>;
-  desc.setter = This::WrapSetter(This::SetterTag<setter>());
+  desc.setter =
+      setter == nullptr ? nullptr : details::TemplatedInstanceVoidCallback<T, setter>;
   desc.data = data;
   desc.attributes = attributes;
   return desc;
 }
@@
 template <typename T>
 template <typename InstanceWrap<T>::InstanceGetterCallback getter,
           typename InstanceWrap<T>::InstanceSetterCallback setter>
 inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceAccessor(
     Symbol name, napi_property_attributes attributes, void* data) {
   napi_property_descriptor desc = napi_property_descriptor();
   desc.name = name;
   desc.getter = details::TemplatedInstanceCallback<T, getter>;
-  desc.setter = This::WrapSetter(This::SetterTag<setter>());
+  desc.setter =
+      setter == nullptr ? nullptr : details::TemplatedInstanceVoidCallback<T, setter>;
   desc.data = data;
   desc.attributes = attributes;
   return desc;
 }
```

I do not have access to a Windows Server 2025 machine, but you can try pushing these changes (node-gyp version and the diff) and rerun the actions to see if it will be fixed.

--- a/.github/workflows/ci-win.yml
+++ b/.github/workflows/ci-win.yml
@@ -60,6 +60,13 @@ jobs:
     - name: Install dependencies
       run: |
         npm install
+    # node-gyp@12 (from package.json) supports Visual Studio 2026, but only
+    # node-gyp@13 emits the linker options that Node.js 26 builds require
+    # (older node-gyp trips LNK1117 on '/opt:lldltojobs'). Upgrade in place for
+    # Node.js >= 26; other versions keep node-gyp@12.
+    - name: Use node-gyp@13 for Node.js >= 26
+      if: matrix.node-version == '26.x'
+      run: npm install --no-save node-gyp@13
     - name: npm test
       shell: bash
       run: |

--- a/.github/workflows/ci-win.yml
+++ b/.github/workflows/ci-win.yml
@@ -10,7 +10,7 @@ permissions:
 
 jobs:
   test:
-    timeout-minutes: 30
+    timeout-minutes: 60
     strategy:
       fail-fast: false
       matrix:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -55,6 +55,11 @@ jobs:
     - name: Install dependencies
       run: |
         npm install
+    # Node.js >= 26 requires node-gyp@13; older versions keep node-gyp@12
+    # (from package.json).
+    - name: Use node-gyp@13 for Node.js >= 26
+      if: matrix.node-version == '26.x'
+      run: npm install --no-save node-gyp@13
     - name: npm test
       run: |
         if [ "${{ matrix.api_version }}" = "experimental" ]; then

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -10,7 +10,7 @@ permissions:
 
 jobs:
   test:
-    timeout-minutes: 30
+    timeout-minutes: 60
     strategy:
       fail-fast: false
       matrix:

--- a/napi-inl.h
+++ b/napi-inl.h
@@ -4681,48 +4681,71 @@ template <typename T>
 template <typename InstanceWrap<T>::InstanceVoidMethodCallback method>
 inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceMethod(
     const char* utf8name, napi_property_attributes attributes, void* data) {
+#ifdef _MSC_VER
+  // MSVC (as of v145 / Visual Studio 2026) raises an internal compiler error
+  // (C1001) when a pointer-to-member-function is used as a non-type template
+  // parameter, as the static compile-time dispatch below does. On MSVC, fall
+  // back to the runtime overload, which passes `method` as a value instead.
+  return InstanceMethod(utf8name, method, attributes, data);
+#else
   napi_property_descriptor desc = napi_property_descriptor();
   desc.utf8name = utf8name;
   desc.method = details::TemplatedInstanceVoidCallback<T, method>;
   desc.data = data;
   desc.attributes = attributes;
   return desc;
+#endif
 }
 
 template <typename T>
 template <typename InstanceWrap<T>::InstanceMethodCallback method>
 inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceMethod(
     const char* utf8name, napi_property_attributes attributes, void* data) {
+#ifdef _MSC_VER
+  // See the note in the InstanceMethod overload above.
+  return InstanceMethod(utf8name, method, attributes, data);
+#else
   napi_property_descriptor desc = napi_property_descriptor();
   desc.utf8name = utf8name;
   desc.method = details::TemplatedInstanceCallback<T, method>;
   desc.data = data;
   desc.attributes = attributes;
   return desc;
+#endif
 }
 
 template <typename T>
 template <typename InstanceWrap<T>::InstanceVoidMethodCallback method>
 inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceMethod(
     Symbol name, napi_property_attributes attributes, void* data) {
+#ifdef _MSC_VER
+  // See the note in the InstanceMethod overload above.
+  return InstanceMethod(name, method, attributes, data);
+#else
   napi_property_descriptor desc = napi_property_descriptor();
   desc.name = name;
   desc.method = details::TemplatedInstanceVoidCallback<T, method>;
   desc.data = data;
   desc.attributes = attributes;
   return desc;
+#endif
 }
 
 template <typename T>
 template <typename InstanceWrap<T>::InstanceMethodCallback method>
 inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceMethod(
     Symbol name, napi_property_attributes attributes, void* data) {
+#ifdef _MSC_VER
+  // See the note in the InstanceMethod overload above.
+  return InstanceMethod(name, method, attributes, data);
+#else
   napi_property_descriptor desc = napi_property_descriptor();
   desc.name = name;
   desc.method = details::TemplatedInstanceCallback<T, method>;
   desc.data = data;
   desc.attributes = attributes;
   return desc;
+#endif
 }
 
 template <typename T>
@@ -4768,6 +4791,10 @@ template <typename InstanceWrap<T>::InstanceGetterCallback getter,
           typename InstanceWrap<T>::InstanceSetterCallback setter>
 inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceAccessor(
     const char* utf8name, napi_property_attributes attributes, void* data) {
+#ifdef _MSC_VER
+  // See the note in the InstanceMethod overload above.
+  return InstanceAccessor(utf8name, getter, setter, attributes, data);
+#else
   napi_property_descriptor desc = napi_property_descriptor();
   desc.utf8name = utf8name;
   desc.getter = details::TemplatedInstanceCallback<T, getter>;
@@ -4775,6 +4802,7 @@ inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceAccessor(
   desc.data = data;
   desc.attributes = attributes;
   return desc;
+#endif
 }
 
 template <typename T>
@@ -4782,6 +4810,10 @@ template <typename InstanceWrap<T>::InstanceGetterCallback getter,
           typename InstanceWrap<T>::InstanceSetterCallback setter>
 inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceAccessor(
     Symbol name, napi_property_attributes attributes, void* data) {
+#ifdef _MSC_VER
+  // See the note in the InstanceMethod overload above.
+  return InstanceAccessor(name, getter, setter, attributes, data);
+#else
   napi_property_descriptor desc = napi_property_descriptor();
   desc.name = name;
   desc.getter = details::TemplatedInstanceCallback<T, getter>;
@@ -4789,6 +4821,7 @@ inline ClassPropertyDescriptor<T> InstanceWrap<T>::InstanceAccessor(
   desc.data = data;
   desc.attributes = attributes;
   return desc;
+#endif
 }
 
 template <typename T>

--- a/package.json
+++ b/package.json
@@ -424,6 +424,7 @@
     "eslint": "^9.13.0",
     "fs-extra": "^11.1.1",
     "neostandard": "^0.12.0",
+    "node-gyp": "^12.4.0",
     "pre-commit": "^1.2.2",
     "semver": "^7.6.0"
   },

